File: DSA/HarshRaj_590025597/Day_4/Question2.py
def rotate(arr, k):
    n = len(arr)
    if n == 0:
        return arr
    k = k % n

    def reverse(a, left, right):
        while left < right:
            a[left], a[right] = a[right], a[left]
            left += 1
            right -= 1

    reverse(arr, 0, n - 1)
    reverse(arr, 0, k - 1)
    reverse(arr, k, n - 1)
    return arr


arr = list(map(int, input().split(',')))
k = int(input())
result = rotate(arr, k)
print(result)

File: DSA/Mayank_Kumar_590012362/Day_4/Question2.java
public class Question2 {
    public void rev(int[] arr, int l, int r) {
        while (l < r) {
            int t = arr[l]; arr[l++] = arr[r]; arr[r--] = t;
        }
    }
    private void rotate(int[] arr, int k) {
        int n = arr.length; k = k % n;
        rev(arr, 0, n - 1);
        rev(arr, 0, k - 1);
        rev(arr, k, n - 1);
    }
}
